akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala

```diff
@@ -145,7 +140,8 @@ private[akka] class InputStreamAdapter(sharedBuffer: BlockingQueue[StreamToAdapt
   }
 
   private[this] def readBytes(a: Array[Byte], begin: Int, length: Int): Int = {
-    val availableInChunk = detachedChunk.size - skipBytes
+    require(detachedChunk.isInstanceOf[Some[ByteString]], "Chunk must be pulled from shared buffer")
```

Comment:
I don't think this should be more optimizaed than if check,maybe just follow his heart.